adaptive_breakpoints/lib/src/adaptive_breakpoints.dart

```diff
@@ -0,0 +1,300 @@
+// Copyright 2020, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+import 'package:flutter/material.dart';
+
+/// Adaptive Window in Material has five different window sizes. Each window size
+/// represents a range of devices.
+///
+/// Extra small represents phones and small tablets in portrait view.
+/// Small represents tablets in portrait view and phones in landscape view.
+/// Medium represents large tablets in landscape view.
+/// Large represents computer screens.
+/// Extra large represents large computer screens.
+///
+/// https://material.io/design/layout/responsive-layout-grid.html#breakpoints
+enum AdaptiveWindowType {
+  xs,
+  s,
+  m,
+  l,
+  xl,
+}
+
+/// This extension lets you access variables associated with each window type.
+extension AdaptiveWindowTypeExtension on AdaptiveWindowType {
+  String get name {
+    switch (this) {
+      case AdaptiveWindowType.xs:
+        return 'xsmall';
+      case AdaptiveWindowType.s:
+        return 'small';
+      case AdaptiveWindowType.m:
+        return 'medium';
+      case AdaptiveWindowType.l:
+        return 'large';
+      case AdaptiveWindowType.xl:
+        return 'xlarge';
+      default:
+        throw AssertionError('Error');
+    }
+  }
+
+  RangeValues get widthRangeValues {
+    switch (this) {
+      case AdaptiveWindowType.xs:
+        return RangeValues(0, 599);
+      case AdaptiveWindowType.s:
+        return RangeValues(600, 1023);
+      case AdaptiveWindowType.m:
+        return RangeValues(1024, 1439);
+      case AdaptiveWindowType.l:
+        return RangeValues(1440, 1919);
+      case AdaptiveWindowType.xl:
+        return RangeValues(1920, double.infinity);
+      default:
+        throw AssertionError('Error');
+    }
+  }
+
+  RangeValues get heightLandscapeRangeValues {
+    switch (this) {
+      case AdaptiveWindowType.xs:
+        return RangeValues(0, 359);
+      case AdaptiveWindowType.s:
+        return RangeValues(360, 719);
+      case AdaptiveWindowType.m:
+        return RangeValues(720, 959);
+      case AdaptiveWindowType.l:
+        return RangeValues(960, 1279);
+      case AdaptiveWindowType.xl:
+        return RangeValues(1280, double.infinity);
+      default:
+        throw AssertionError('Error');
+    }
+  }
+
+  RangeValues get heightPortraitRangeValues {
+    switch (this) {
+      case AdaptiveWindowType.xs:
+        return RangeValues(0, 959);
+      case AdaptiveWindowType.s:
+        return RangeValues(360, 1599);
+      case AdaptiveWindowType.m:
+        return RangeValues(720, 1919);
+      case AdaptiveWindowType.l:
+        return RangeValues(1920, double.infinity);
+      case AdaptiveWindowType.xl:
+        return RangeValues(1920, double.infinity);
+      default:
+        throw AssertionError('Error');
+    }
+  }
+}
+
+class AdaptiveWindowLimits {
```

Comment:
I believe this isn't actually used anywhere and it's still missing documentation. Could you delete it?

Reply:
It isn't used yet but it is for a future idea. I deleted it since it isn't being used right now.